src/main/java/com/avaje/ebeaninternal/server/core/PersistRequestBean.java

```diff
@@ -787,10 +787,14 @@ private void controllerPost() {
         controller.postInsert(this);
         break;
       case UPDATE:
+      case SOFT_DELETE:
+    	  /*
+    	   * In the previous version, in case SOFT_DELETE the method controller.postDelete was called. However,
+    	   * this lead to an inconsistent situation, because first controller.preUpdate is called.
+    	   */
```

Comment:
preUpdate/postUpdate makes more sense, as softdelete is no real delete. (and undeletion is done by update, too)
maybe we add a preSoftDelete/postSoftDelete event later.